tests/test_dupefilters.py

```diff
@@ -6,10 +6,31 @@
 from scrapy.dupefilters import RFPDupeFilter
 from scrapy.http import Request
 from scrapy.utils.python import to_bytes
+from scrapy.utils.job import job_dir
+from scrapy.utils.test import get_crawler
 
 
 class RFPDupeFilterTest(unittest.TestCase):
 
+    def test_dupefilter_from_crawler(self):
+
+        class FromCrawlerRFPDupeFilter(RFPDupeFilter):
+
+            @classmethod
+            def from_crawler(cls, crawler):
+                debug = crawler.settings.getbool('DUPEFILTER_DEBUG')
+                df = cls(job_dir(crawler.settings), debug)
+                df.user_agent = crawler.settings.get('USER_AGENT')
+                return df
+
+        crawler = get_crawler(settings_dict={'DUPEFILTER_DEBUG': True, 'USER_AGENT': 'test ua'})
+        dupefilter = FromCrawlerRFPDupeFilter.from_crawler(crawler)
```

Comment:
This test doesn't check that Scheduler calls from_crawler method. I think this tests would pass even before all changes in this PR, so it is not really testing the new feature.

Reply:
You're right, but to be fair, the current tests don't check the way the Scheduler creates the dupefilter either :-)

I can add a test for that.